Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -426,17 +222,114 @@ zsh: # . ~/.zshrc
+
+##### Debugging with lldb
+
+iOS ships with the console app debugserver, which allows remote debugging via gdb or lldb. By default, however, debugserver can't be used to attach to arbitrary processes (it is usually used only for debugging self-developed apps deployed with Xcode). To enable debugging of third-party apps, the `task_for_pid` entitlement must be added to the debugserver executable. An easy way to do this is to add the entitlement to the [debugserver binary shipped with Xcode](http://iphonedevwiki.net/index.php/Debugserver "Debug Server on the iPhone Dev Wiki").
+
+To obtain the executable, mount the following DMG image:
 
 ```shell
-$ curl -O https://build.frida.re/frida/ios/lib/FridaGadget.dylib
+/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/DeviceSupport/<target-iOS-version>/DeveloperDiskImage.dmg
+```
+
+You'll find the debugserver executable in the `/usr/bin/` directory on the mounted volume. Copy it to a temporary directory, then create a file called `entitlements.plist` with the following content:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/ PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+ <key>com.apple.springboard.debugapplications</key>
+ <true/>
+ <key>run-unsigned-code</key>
+ <true/>
+ <key>get-task-allow</key>
+ <true/>
+ <key>task_for_pid-allow</key>
+ <true/>
+</dict>
+</plist>
+```
+
+Apply the entitlement with codesign:
+
+```shell
+$ codesign -s - --entitlements entitlements.plist -f debugserver
+```
+
+Copy the modified binary to any directory on the test device. The following examples use usbmuxd to forward a local port through USB.
+
+```shell
+$ ./tcprelay.py -t 22:2222
+$ scp -P2222 debugserver root@localhost:/tmp/
+```
+
+You can now attach debugserver to any process running on the device.
+
+```shell
+VP-iPhone-18:/tmp root# ./debugserver *:1234 -a 2670
+debugserver-@(#)PROGRAM:debugserver  PROJECT:debugserver-320.2.89
+for armv7.
+Attaching to process 2670...
 ```
 
-We'll be using standard tools that come with macOS and Xcode in addition to the tools mentioned above. Make sure you have the [Xcode command line developer tools](https://railsapps.github.io/xcode-command-line-tools.html "Xcode Command Line Tools") installed.
+#### Tracing
+
+##### Execution Tracing
+
+Intercepting Objective-C methods is a useful iOS security testing technique. For example, you may be interested in data storage operations or network requests. In the following example, we'll write a simple tracer for logging HTTP(S) requests made via iOS standard HTTP APIs. We'll also show you how to inject the tracer into the Safari web browser.
+
+In the following examples, we'll assume that you are working on a jailbroken device. If that's not the case, you first need to follow the steps outlined in the previous section to repackage the Safari app.
```

Comment:
perhaps refer to the section at hand?

Reply:
actually to section "Repackaging and Re-Signing", done ;)